microros_olimex_e407_extensions/Src/custom_memory_manager.c

```diff
@@ -300,21 +300,27 @@ void *pvPortRealloc( void *pv, size_t xWantedSize )
 
 	void * newmem = pvPortMalloc(xWantedSize);
 
-	uint8_t *puc = ( uint8_t * ) pv;
-	BlockLink_t *pxLink;
+	if (newmem != NULL && pv != NULL)
+	{
+		uint8_t *puc = ( uint8_t * ) pv;
+		BlockLink_t *pxLink;
 
-	puc -= xHeapStructSize;
-	pxLink = ( void * ) puc;
+		puc -= xHeapStructSize;
+		pxLink = ( void * ) puc;
 
+		char *in_src = (char*)pv;
+		char *in_dest = (char*)newmem;
 
-	char *in_src = (char*)pv;
-  	char *in_dest = (char*)newmem;
-	size_t count = pxLink->xBlockSize & ~xBlockAllocatedBit;
+		size_t count = (pxLink->xBlockSize & ~xBlockAllocatedBit) - xHeapStructSize;
+		if (xWantedSize < count)
+		{
+			count = xWantedSize;
+		}
```

Comment:
@Acuadros95, maybe it is not my place to comment about formatting (sorry for nitpicking :grimacing:), but would it make sense to convert this `if` statement to a one-liner like the `while` statement below? Or the other way around, add the brackets to the `while` statement.

Reply:
No problem: 7f9eb5371ad284f47a004e500cea5c196598e2ce